### DISCUSSION TITLE
Making my own release of Checkstyle for learning

### DISCUSSION THREAD

**Discussion by Rahulkhinchi03** [checkstyle/checkstyle#11648] [Show and tell]
I am opening this discussion to ask for guidance if I am stuck in making a release of Checkstyle for learning purposes.

@romani I am following your wiki here: https://github.com/sevntu-checkstyle/dsm-maven-plugin/wiki/How-to-config-GPG-and-sign-artifact-with-it

I set up a Jira account: https://issues.sonatype.org/browse/OSSRH-80906
Public Key: https://keyserver.ubuntu.com/pks/lookup?search=0xE407C4F2&fingerprint=on&op=index

```
rahul@RK7:~$ gpg --list-secret-keys --keyid-format 0xshort
/home/rahul/.gnupg/pubring.kbx
------------------------------
pub   rsa3072/0xE407C4F2 2022-06-12 [SC] [expires: 2024-06-11]
      85D1D009DE62172B4887A41AE96D58DDE407C4F2
uid           [ultimate] Rahul Khinchi <rahulkhinchirk7@gmail.com>
sub   rsa3072/0x1BB83448 2022-06-12 [E] [expires: 2024-06-11]

```

I added a plugin to the build section.

```
          <plugin>
            <groupId>org.apache.maven.plugins</groupId>
            <artifactId>maven-gpg-plugin</artifactId>
            <version>1.4</version>
            <executions>
              <execution>
                <id>sign-artifacts</id>
                <phase>verify</phase>
                <goals>
                  <goal>sign</goal>
                </goals>
              </execution>
            </executions>
          </plugin>
```
```
gpg --default-key 0xE407C4F2 --clearsign
gpg: using "0xE407C4F2" as default secret key for signing
```

Problem: I am not able to sign the artifact. When I am running `mvn clean verify -Dgpg.passphrase=rahulrk7`, 
ERROR: 
```
gpg: no default secret key: No secret key
gpg: signing failed: No secret key
```

How can I sign the artifact with, 
`$gpg -ab artifact.jar` , What should I use in this command?

**Comment by romani** [comment]
dealing with gpg keys , might be problematic, as you would need to actually deploy to some secure environment.
Lets focus for now on automation of phases and later on secutrity and secrets storage.
